### PR TITLE
Update normalization.py

### PR DESCRIPTION
**Pull request by dongdongunique**
Before update, the class has two bugs

1) torch.chunk(emb,2) should torch.chunk(emb,2,dim=1), otherwise, it will split the batch instead of the feature space.

2) scale=scale.unsqueeze(1).expand_as(x)
     shift=shift.unsqueeze(1).expand_as(x); otherwise, the scale and shift could not broadcast to the x.shape, and consequently it will throw bugs

# What does this PR do?

Fixes # (issue)

## Before submitting
-   This PR fixes two bugs
-   Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
-   Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
-  [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
-  [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
-   Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.
@sayakpaul @yiyixuxu @DN6

### CODE REVIEW

File: src/diffusers/models/normalization.py [hunk 1]
```diff
@@ -43,7 +43,9 @@ def __init__(self, embedding_dim: int, num_embeddings: int):
 
     def forward(self, x: torch.Tensor, timestep: torch.Tensor) -> torch.Tensor:
         emb = self.linear(self.silu(self.emb(timestep)))
-        scale, shift = torch.chunk(emb, 2)
+        scale, shift = torch.chunk(emb, 2,dim=1)
```
Comment (kashif) [inline src/diffusers/models/normalization.py:46]:
```suggestion
        scale, shift = torch.chunk(emb, 2, dim=1)
```

**Comment by kashif** [comment]
@sayakpaul good catch! In the orignal method, they chunk it in `dim=2` if i recall... https://github.com/colorful-liyu/3DQD/blob/1c616db20ae3111ee98ecf4fb23be3c1f9fa66d4/models/networks/transformer_networks/diff_transformer/transformer_utils.py#L136

**Comment by dongdongunique** [comment]
> @sayakpaul good catch! In the orignal method, they chunk it in `dim=2` if i recall... https://github.com/colorful-liyu/3DQD/blob/1c616db20ae3111ee98ecf4fb23be3c1f9fa66d4/models/networks/transformer_networks/diff_transformer/transformer_utils.py#L136

Yes, in the transformer, they chunk it with dim=2, but they can do it because they have unsqueeze(1), so there are no conflicts between the transformer and my PR. 
image

**Comment by sayakpaul** [comment]
@kashif possible for you to run the affected tests?

**Comment by kashif** [comment]
the failing test is due to formatting i believe... let me fix it

File: src/diffusers/models/normalization.py (hunk 1 above)
Comment (kashif) [inline src/diffusers/models/normalization.py:46]:
```suggestion
        scale, shift = torch.chunk(emb, 2, dim=1)
```

**Comment by sayakpaul** [comment]
No I mean any other pipeline or model level tests that might have been affected by this?

**Comment by kashif** [comment]
ok

File: src/diffusers/models/normalization.py [hunk 2]
```diff
@@ -43,7 +43,9 @@ def __init__(self, embedding_dim: int, num_embeddings: int):
 
     def forward(self, x: torch.Tensor, timestep: torch.Tensor) -> torch.Tensor:
         emb = self.linear(self.silu(self.emb(timestep)))
-        scale, shift = torch.chunk(emb, 2)
+        scale, shift = torch.chunk(emb, 2, dim=1)
+        scale=scale.unsqueeze(1).expand_as(x)
```
Comment (kashif) [inline src/diffusers/models/normalization.py:47]:
```suggestion
        scale = scale.unsqueeze(1).expand_as(x)
```

**Comment by kashif** [comment]
@sayakpaul so i believe only VQ-diffusion was using this normalization layer so going to check that

**Comment by dongdongunique** [comment]
> @sayakpaul so i believe only VQ-diffusion was using this normalization layer so going to check that

I encountered this problem when I tried to use Transformer2DModel for unconditional image generation. Maybe this class should be checked too.

**Comment by kashif** [comment]
@dongdongunique so by default the norm for Transformer2D via `norm_type="layer_norm"` so you mean you were trying with your own and setting the norm type to `ada_norm`?

**Comment by dongdongunique** [comment]
> @dongdongunique so by default the norm for Transformer2D via `norm_type="layer_norm"` so you mean you were trying with your own and setting the norm type to `ada_norm`?

Yes

**Review by kashif — APPROVED** [review]

**Comment by dongdongunique** [comment]
I think i found where the bug is:

If I tried to pass the timestep like this
image
the emb will only have one dimension and throw the bug 
image
image

so I figured out a way to fix this bug, but I wonder if this fix will be robust?
image

**Comment by kashif** [comment]
yeah lets make it robust as the way this is used is when there are a batch of time indices... rather than a tensor of them... I would say either unsqueeze on your end and perhaps document the input shape in the norm, or make it robust to different input shapes?

**Comment by dongdongunique** [comment]
I checked the DiT's code 
https://github.com/facebookresearch/DiT/blob/main/models.py#L237
they unsqueeze the timestep before passing it through the forward;
Perhaps we can follow them?

**Comment by kashif** [comment]
@dongdongunique  yes I believe that is the convention in all the normalization layers taking the time index... so I would say do the same on your side and perhaps close this issue for now?

**Comment by dongdongunique** [comment]
> @dongdongunique yes I believe that is the convention in all the normalization layers taking the time index... so I would say do the same on your side and perhaps close this issue for now?

ok!

**Comment by dongdongunique** [comment]
wait a minute, I recall in the tutorial of diffusers: https://huggingface.co/docs/diffusers/main/en/using-diffusers/write_own_pipeline
image
they did not unsqueeze the timestep on their end, this code will encounter the same problem if they use the Transformer2D with ada_norm

**Pull request closed without merging**